src/Umbraco.Web.UI.Client/src/packages/core/collection/default/collection-default.element.ts

```diff
@@ -98,9 +107,10 @@ export class UmbCollectionDefaultElement extends UmbLitElement {
 
 	#renderEmptyState() {
 		if (!this._isDoneLoading) return nothing;
+
 		return html`
 			<div id="empty-state" class="uui-text">
-				<h4><umb-localize key="collection_noItemsTitle"></umb-localize></h4>
+				<h4>${this.#generateEmptyStateLabel()}</h4>
```

Comment:
Then instead of retrieving this everytime we render, we instead should store it in a @State property, so we only do the logic once.